buildpacks/gradle/src/framework.rs

```diff
@@ -1,26 +1,89 @@
 use crate::gradle_command::GradleDependencyReport;
+use buildpacks_jvm_shared::fs::list_directory_contents;
+use libcnb::data::launch::{Process, ProcessBuilder};
+use libcnb::data::process_type;
+use std::path::Path;
 
 pub(crate) fn detect_framework(dependency_report: &GradleDependencyReport) -> Option<Framework> {
-    DEPENDENCY_TO_FRAMEWORK_MAPPINGS
-        .into_iter()
-        .find_map(|(group_id, artifact_id, framework)| {
+    DEPENDENCY_TO_FRAMEWORK_MAPPINGS.into_iter().find_map(
+        |(configuration, group_id, artifact_id, framework)| {
             dependency_report
-                .contains_dependency("runtimeClasspath", group_id, artifact_id)
+                .contains_dependency(configuration, group_id, artifact_id)
                 .then_some(framework)
-        })
+        },
+    )
+}
+
+#[allow(clippy::case_sensitive_file_extension_comparisons)]
```

Comment:
I decided to rather add this `#[allow]` than to allow it globally. The lint can be useful, even when it's not in this case here.